src/tools/cephfs/cephfs-shell

```diff
@@ -243,25 +244,58 @@ def copy_to_local(shell, remote_path, local_path):
         fd.close()
 
 
-def dirwalk(dir_name):
+def dirwalk(dir_name, giveDirs = 0):
     """
     walk a directory tree, using a generator
     """
     dir_name = os.path.normpath(dir_name)
     for item in list_items(dir_name)[2:]:
         fullpath = os.path.join(dir_name, item.d_name.decode('utf-8'))
-        yield fullpath.rsplit('/', 1)[0] + '/'
         if is_dir_exists(item.d_name, fullpath.rsplit('/', 1)[0]):
             if not len(list_items(fullpath)[2:]):
-                yield fullpath
+                yield os.path.normpath(fullpath)
             else:
-                return dirwalk(fullpath)
+                for x in dirwalk(fullpath):
+                    if is_dir_exists(x):
+                        if giveDirs:
+                            yield x 
+                    else:
+                        yield x 
         else:
-            yield fullpath
+            yield os.path.normpath(fullpath)
 
 
 class CephFSShell(Cmd):
 
+    try:
+        from cmd2 import with_argparser
+    except ImportError:
+        def with_argparser(self, argparser):
+            import functools
+            def argparser_decorator(func):
+                @functools.wraps(func)
+                def wrapper(thiz, cmdline):
+                    # do not split if it's already a list
+                    if not isinstance(cmdline, list):
+                        arglist = shlex.split(cmdline, posix=False)
+                        # in case user quotes the command args
+                        arglist = [arg.strip('\'""') for arg in arglist]
```

Comment:
```py
if isinstance(cmdline, list):
  arglist = cmdline
else:
  // ... split and strip
```

and i'm inlined to move `with_argparser()` out from `CephFSShell`, as the function signature of `with_argparser(self, argparser)` is different from that of `cmd.with_argparser(argparser)`, also, less indent is easier to understand.

Reply:
@rpavani1998 this comment is not addressed.